Algorithmia/CLI.py

```diff
@@ -0,0 +1,233 @@
+					print(error)
+
+			elif(inputs[0] == '-t'):
+				#text
+				algo_input = inputs[1]
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input.encode('utf-8'),
+					headers={'Authorization':key.read(),'Content-Type':'text/plain'}, params= algo.query_parameters).json())
+				key.close()
+
+			elif(inputs[0] == "-j"):
+				#json
+				algo_input = inputs[1]
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input,
+					headers={'Authorization':key.read(),'Content-Type':'application/json'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-b"):
+				#binary
+				algo_input = inputs[1]
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=bytes(algo_input),
+					headers={'Authorization':key.read(),'Content-Type':'application/octet-stream'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-D" or inputs[0] == "--data-file"):
+				#data file
+				algo_input = open(inputs[1],"r").read()
+				try:
+					result = algo.pipe(algo_input)
+				except Exception as error:
+					print(error)
+
+			elif(inputs[0] == "-T"):
+				#text file
+				algo_input = open(inputs[1],"r").read()
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input.encode('utf-8'),
+					headers={'Authorization':key.read(),'Content-Type':'text/plain'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-J"):
+				#json file
+				#read json file and run algo with that input bypassing the auto detection of input type in pipe
+				algo_input = open(inputs[1],"r").read()
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=json.dumps(algo_input).encode('utf-8'),
+					headers={'Authorization':key.read(),'Content-Type':'application/json'}, params= algo.query_parameters))
+				key.close()
+
+			elif(inputs[0] == "-B"):
+				#binary file
+				algo_input = open(inputs[1],"rb").read()
+				print(algo_input)
+				print("end input \n")
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=bytes(algo_input),
+					headers={'Authorization':key.read(),'Content-Type':'application/octet-stream'}, params= algo.query_parameters).json())
+				key.close()
+
+			else:
+				algo_input = inputs[0]
+				try:
+					result = algo.pipe(algo_input)
+				except Exception as error:
+					print(error)
+			
+			#handle output flags
```

Comment:
as a general comment, I'm not sure how important the `response` and `response-body` output options are (I mostly wanted them to help me debug authoring the CLI itself and occasionally debug the server response without needing to find the right `curl` command. @platypii can make the call on that.)

But I do think `--debug` `--silence` and maybe `-o <file>` tend to be pretty useful for a variety of use cases.